lib/puppet/pops/types/types.rb

```diff
@@ -840,11 +1021,26 @@ def generalize
       def hashed_elements
         @hashed ||= @elements.reduce({}) {|memo, e| memo[e.name] = e; memo }
       end
- 
+
       def hash
         @elements.hash
       end
 
+      def iterable?
+        true
+      end
+
+      def iterable_type
+        if self == DEFAULT
+          PIterableType.new(PTupleType.new([PAnyType::DEFAULT], PHashType::TUPLE_SIZE))
+        else
+          tc = TypeCalculator.singleton
+          key_type = tc.infer_and_reduce_type(@elements.map {|se| se.key_type })
```

Comment:
Oversight. This should be changed to use the same improved logic that was added to Iterable.